rdkit/Chem/EState/EState.py

```diff
@@ -45,28 +44,26 @@ def EStateIndices(mol, force=True):
 
   tbl = Chem.GetPeriodicTable()
   nAtoms = mol.GetNumAtoms()
-  Is = numpy.zeros(nAtoms, numpy.float)
+  Is = [0.0] * nAtoms
   for i in range(nAtoms):
     at = mol.GetAtomWithIdx(i)
-    atNum = at.GetAtomicNum()
     d = at.GetDegree()
     if d > 0:
-      h = at.GetTotalNumHs()
-      dv = tbl.GetNOuterElecs(atNum) - h
+      atNum = at.GetAtomicNum()
+      dv = tbl.GetNOuterElecs(atNum) - at.GetTotalNumHs()
       N = GetPrincipleQuantumNumber(atNum)
       Is[i] = (4. / (N * N) * dv + 1) / d
-  dists = Chem.GetDistanceMatrix(mol, useBO=0, useAtomWts=0)
-  dists += 1
-  accum = numpy.zeros(nAtoms, numpy.float)
+  dists = Chem.GetDistanceMatrix(mol, useBO=0, useAtomWts=0) + 1
+  
+  accum = [0.0] * nAtoms
   for i in range(nAtoms):
     for j in range(i + 1, nAtoms):
       p = dists[i, j]
       if p < 1e6:
         tmp = (Is[i] - Is[j]) / (p * p)
         accum[i] += tmp
         accum[j] -= tmp
-
-  res = accum + Is
+  res = numpy.add(accum, Is, dtype='float')
```

Comment:
This will convert the `accum` and `Is` lists into numpy arrays. It therefore is better to leave the original definition of these arrays using `numpy.zeros` . You can then leave the old statement `res = accum + Is`.

Reply:
I don't agree with it. 
Yes, the main reason I make it here due to the element-wise access to `np.array` which is not recommended in numpy.

Reply:
If np.add is more efficient and recommended, then use it. However, do you think that using numpy arrays for `accum` and `Is` in the code above is more efficient?

Reply:
I am so sorry about that, but to your question, at least it is. At current, I did not make any strong optimization, 
If you want, I would check it in the near future.

Reply:
I believe a greater option is to make heavy optimization.
I can merge `accum` and `Is` at once. 
1) Initialize `res` as a python list
2) Calculate
3) Switch back to numpy.
Does it sound good?
@gedeck